FileSystemSimulator/IOSystem/IOSystem.cpp

```diff
@@ -0,0 +1,52 @@
+#include "stdafx.h"
+#include "IOSystem.h"
+
+
+IOSystem::IOSystem(Disk* ldisk)
+{
+	this->ldisk = ldisk;
+}
+
+
+IOSystem::~IOSystem()
+{
+}
+
+int * IOSystem::getBlockLocationOnDisk(int blockNumber)
+{
+	int result[3];
+	int cylinderNumber = -1, trackNumber = -1, sectorNumber = -1;
+
+	cylinderNumber = blockNumber / BLOCKS_IN_CYLINDER;
+	trackNumber = (blockNumber % BLOCKS_IN_CYLINDER) / Track::NUMBER_OF_SECTORS;
+	sectorNumber = (blockNumber % BLOCKS_IN_CYLINDER) % Track::NUMBER_OF_SECTORS;
+
+	result[0] = cylinderNumber;
+	result[1] = trackNumber;
+	result[2] = sectorNumber;
+	return result;
```

Comment:
you are returning a pointer to local variable _result_. When quit the scope, all the local variables will be destroyed, so poiner will be point at garbage. The right way is creating array on heap
```suggestion
	int* result = new int[3]; 
```
but dont forget to free memory after using 
```suggestion
	int* blockLocation = getBlockLocationOnDisk(blockNumber);
	... // using of blockLocation
	delete[] blockLocation;
```
or use std::vector (more safe)

Reply:
это не вместо return result; если что